File: gm-utils/js/src/commom/utils/object/UJson.js
import Null from './Null';
import StringEmpty from '../string/StringEmpty';

export default class UJson {

	static getKey(key) {
		return "\""+key+"\": ";
	}

	static getNull(key) {
		return UJson.getKey(key) + "null,";
	}

	static itemString(key, value) {

		if (StringEmpty.is(value)) {
			return UJson.getNull(key);
		}
		return UJson.getKey(key) + "\""+value+"\",";

	}

	static itemInteger(key, value) {

		if (Null.is(value)) {
			return UJson.getNull(key);
		}
		return UJson.getKey(key) + value+",";

	}

	static itemLong(key, value) {

		if (Null.is(value)) {
			return UJson.getNull(key);
		}
		return UJson.getKey(key) + value+",";

	}

	static itemBoolean(key, value) {

		if (Null.is(value)) {
			return UJson.getNull(key);
		}
		return UJson.getKey(key) + value+",";

	}
	static itemObj(key, value) {

		if (Null.is(value)) {
			return UJson.getNull(key);
		}
		return UJson.getKey(key) + value.toJSON()+",";

	}

}

File: gm-utils/src/main/java/src/commom/utils/object/UJson.java
package src.commom.utils.object;

import src.commom.utils.string.StringEmpty;

public class UJson {

	private static String getKey(String key) {
		return "\""+key+"\": ";
	}

	private static String getNull(String key) {
		return getKey(key) + "null,";
	}

	public static String itemString(String key, String value) {
		
		if (StringEmpty.is(value)) {
			return getNull(key);
		}
		
		return getKey(key) + "\""+value+"\",";
		
	}

	public static String itemInteger(String key, Integer value) {

		if (Null.is(value)) {
			return getNull(key);
		}
		
		return getKey(key) + value+",";

	}

	public static String itemLong(String key, Long value) {

		if (Null.is(value)) {
			return getNull(key);
		}
		
		return getKey(key) + value+",";

	}

	public static String itemBoolean(String key, Boolean value) {

		if (Null.is(value)) {
			return getNull(key);
		}
		
		return getKey(key) + value+",";

	}
	
	public static String itemObj(String key, ObjJs value) {

		if (Null.is(value)) {
			return getNull(key);
		}
		
		return getKey(key) + value.toJSON()+",";

	}

}
